torch/csrc/utils/tensor_new.cpp

```diff
@@ -463,13 +470,16 @@ Tensor tensor_ctor(const Type& type, PyObject* args, PyObject* kwargs) {
   auto r = parser.parse(args, kwargs, parsed_args);
   if (r.idx == 0) {
     bool type_inference = r.isNone(1);
+    // args_requires_grad=True if requires_grad is set to True (requires_grad=True)
+    bool args_requires_grad = r.toBool(3);
     return internal_new_from_data(
                typeWithDefault(r, 1, 2, type),
                r.deviceOptional(2),
                r.pyobject(0),
                true,
                true,
-               type_inference)
+               type_inference,
+               args_requires_grad)
         .set_requires_grad(r.toBool(3));
```

Comment:
nit: set this to args_requires_grad so you don't have to read out the arguments again.